File: leetcode/first_unique_character_in_string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        d = {}
        for string in s:
            if string not in d:
                d[string] = 1
            else:
                d[string] += 1

        index = -1
        for i in range(len(s)):
            if d[s[i]] == 1:
                index = i
                break

        return index

File: leetcode/easy/first_unique_character_in_string.ts
function firstUniqChar(s: string): number {
  let map = new Map();

  for (let v of s) {
    if (v in map) {
      map[v] += 1;
    } else {
      map[v] = 1;
    }
  }

  let index = -1;
  for (let i = 0; i < s.length; i++) {
    if (map[s[i]] == 1) {
      index = i;
      break;
    }
  }

  return index;
}
